orc8r/gateway/c/common/sentry/includes/SentryWrapper.h

```diff
@@ -36,6 +36,11 @@ typedef struct sentry_config {
   bool add_debug_logging;
 } sentry_config_t;
 
+/**
+ * @brief Get the sentry configuration object from shared mconfig
+ */
+void get_sentry_configuration(sentry_config_t* sentry_config);
```

Comment:
`construct_sentry_config_from_mconfig` might be more descriptive. Also, why not just return the struct as a return value rather than using an output parameter? `sentry_config_t get_sentry_configuration();`

Reply:
I have renamed the method. :heavy_check_mark:

Reply:
Thanks for the hint regarding the struct! Your suggestion was also my first thought of implementation. However, as I am coming more from the Python kind of language, we discussed this in the team. Within this discussion I have been advised that the concept of ownership is to be valued in C (as well). We furthermore discussed how memory management is to be taken care of manually in C, and if the objects will be available in different scopes using the different implementations.

My understanding is so far, that
- both approaches would work fine in this case.
- a created entity inside the method would only live outside if it is passed by value with the return statement, which
  - is the case for structs in C by default,
  - is not generally the case for objects in C++.
- The concepts of ownership suggests to create objects where they are used to avoid memory leaks.

From what I got so far of the discussion over the different implementations, my feeling is, that they are matter of taste.
The current implementation being slightly more general (as also working for C++ objects) and more focusing on the ownership principle.
Your suggested implementation being more C specific (using peculiarities of structs) and probably some compiler magic to avoid having two copies of the entity in memory when passing by value.

Did we miss anything, which makes one or the other implementation superior?

Reply:
Yes I think they are both acceptible. I prefer return values as it looks cleaner, but I don't have a strong opinion either way. We do point to the [google C++ guidelines](https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs) for these types of things [(our contributing guidelines](https://docs.magmacore.org/docs/next/contributing/contribute_conventions#c)), so we should follow it unless we have good reasons to do otherwise.

Reply:
Thanks, @themarwhal ! I refactored to follow the Google CPP style guidelines.